### PR TITLE
Public API guard for angularCompilerOptions

### PR DESCRIPTION
**Pull request by alxhub**
Refactor and introduction of a public API guard for `angularCompilerOptions` in tsconfig.

### CODE REVIEW

File: packages/compiler-cli/src/ngtsc/core/api/src/options.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * Whether the compiler should avoid generating code for classes that haven't been exported.
+   * This is only active when building with `enableIvy: true`. Defaults to `true`.
+   */
+  compileNonExportedClasses?: boolean;
+
+  /**
+   * Whether to remove blank text nodes from compiled templates. It is `false` by default starting
+   * from Angular 6.
+   */
+  preserveWhitespaces?: boolean;
```
Comment (JoostK) [inline packages/compiler-cli/src/ngtsc/core/api/src/options.ts]: This was here before this change, but this option is already declared in `LegacyNgOptions`.

File: packages/compiler-cli/src/ngtsc/core/api/src/options.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  /**
+   * Disable TypeScript Version Check.
+   */
+  disableTypeScriptVersionCheck?: boolean;
```
Comment (JoostK) [inline packages/compiler-cli/src/ngtsc/core/api/src/options.ts]: This one should probably be under the public API guard.

File: packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts [hunk 3]
```diff
@@ -6,63 +6,13 @@
...
-export interface ExtendedTsCompilerHost extends ts.CompilerHost, Partial<ResourceHost>,
-    Partial<UnifiedModulesHost> {}
-
 /**
  * Options supported by the legacy View Engine compiler, which are still consumed by the Angular Ivy
  * compiler for backwards compatibility.
  *
  * These are expected to be removed at some point in the future.
+ *
+ * @publicApi
```
Comment (JoostK) [inline packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts]: This is here for the golden generation to include this type, however the type itself is not really public API from a programmatic usage perspective. I'm not sure what can be done about it, but maybe include a disclaimer in the comment.

**Review by JoostK — CHANGES_REQUESTED** [review]
Some minor comments. Also, merging this needs to be coordinated with #35768.

File: packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts (hunk 3 above)
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts]: There is no exported API/documentation generated for these types anyway, so the marker is more "in spirit".

File: packages/compiler-cli/src/ngtsc/core/api/src/options.ts (hunk 1 above)
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/core/api/src/options.ts]: Good catch - deleted.

File: packages/compiler-cli/src/ngtsc/core/api/src/options.ts (hunk 2 above)
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/core/api/src/options.ts]: Done.

File: packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts (hunk 3 above)
Comment (JoostK) [reply packages/compiler-cli/src/ngtsc/core/api/src/public_options.ts]: True, true

**Review by JoostK — APPROVED** [review]
LGTM

**Comment by alxhub** [comment]
[Presubmit](http://test/OCL:299424667:BASE:299424701:1583530293538:c4eb3453)

**Review by IgorMinar — APPROVED** [review]
lgtm for public api changes/tracking. thanks!

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**